cpp/server.cpp

```
@@ -0,0 +1,99 @@
+/**
+ * @brief Redis-like in-memory server with UCall and ustore.
+ */
+#include <cstdio> // `std::printf`
+
+#include "ucall/ucall.h"
+#include "ustore/cpp/db.hpp"
+
+using namespace unum::ustore;
+using namespace unum;
+
+static ustore_database_t db{};
+static ustore_arena_t arena{};
+
+static void set(ucall_call_t call, ucall_callback_tag_t)
+{
+    int64_t *key_ptr{};
+    ucall_str_t value_ptr{};
+    size_t value_len{};
+
+    bool key_found = ucall_param_named_i64(call, "key", 0, key_ptr);
+    bool value_found = ucall_param_named_str(call, "value", 5, &value_ptr, &value_len);
+    if (!key_found || !value_found)
+        return ucall_call_reply_error_invalid_params(call);
+
+    status_t status;
+    ustore_write_t write{};
+    write.db = db;
+    write.error = status.member_ptr();
+    write.arena = &arena;
+    write.tasks_count = 1;
+    write.keys = reinterpret_cast<ustore_key_t *>(key_ptr);
+    write.lengths = reinterpret_cast<ustore_length_t *>(&value_len);
+    write.values = reinterpret_cast<ustore_bytes_cptr_t *>(&value_ptr);
+
+    ustore_write(&write);
+    if (status)
+        return ucall_call_reply_content(call, "OK", 2);
+    return ucall_call_reply_content(call, "", 0);
```

Isn't this a failure? Shouldn't we return an error?